src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -547,16 +547,14 @@ private static Expression[] CreateArguments(ParameterInfo[]? parameters, Factory
 
         var args = new Expression[parameters.Length];
 
-        factoryContext.ArgumentTypes = new Type[parameters.Length];
-        factoryContext.ArgumentExpressions = new Expression[parameters.Length];
-        factoryContext.BoxedArgs = new Expression[parameters.Length];
-        factoryContext.Parameters = new List<ParameterInfo>(parameters);
-
         var hasFilters = factoryContext.FilterFactories is { Count: > 0 };
 
         for (var i = 0; i < parameters.Length; i++)
         {
-            args[i] = CreateArgument(parameters[i], factoryContext);
+            var parameter = parameters[i];
+
+            factoryContext.ParametersAndPropertiesAsParameters.Add(parameter);
```

Comment:
It will change the ordering the metadata (for parameters from `AsParameters`) will be added, right? I don't know if that is a problem and, to be honest, it looks more correct with this change.